Dockerfile

```diff
@@ -0,0 +1,30 @@
+# Build image
+FROM node:alpine
+ARG ENV
+ARG ID
+ARG URL
+ARG COMPONENTS
+
+# Make sure that all the args are not empty
+RUN test -n "$ENV"
+RUN test -n "$ID"
+RUN test -n "$URL"
+RUN test -n "$COMPONENTS"
+
+# Remap args to React env vars
+ENV REACT_APP_ENV=$ENV
+ENV REACT_APP_ID=$ID
+ENV REACT_APP_URL=$URL
+ENV REACT_APP_COMPONENTS=$COMPONENTS
+
+WORKDIR /usr/src/app
+COPY . /usr/src/app/
+
+RUN set -ex; \
+  npm install; \
+  npm run build
+
+# Run image
+FROM nginx:alpine
+WORKDIR /usr/share/nginx/html
+COPY --from=0 /usr/src/app/build/ /usr/share/nginx/html
```

Comment:
One image is used to build the app, a separate one (including nginx) is used to run it. I read that it's the best practice on the internets.

Reply:
Looks strange to me, cannot see the reason for that.

Reply:
It's a strategy to keep the resulting image optimized and without any unnecessary artifacts (like Node engine), see: https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds

And the most important quote from the link above:
> How does it work? The second FROM instruction starts a new build stage with the alpine:latest image as its base. The COPY --from=0 line copies just the built artifact from the previous stage into this new stage.

The final result is still one image, but as tiny as possible, containing only what's needed to run the app.

Reply:
Ah sorry, I only read your comment and thought you are speaking even about about a separate Dockerfile. This of course makes a good sense.